src/main/java/org/checkerframework/checker/noliteral/NoLiteralVisitor.java

```diff
@@ -0,0 +1,230 @@
+package org.checkerframework.checker.noliteral;
+
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ExecutableElement;
+import org.checkerframework.checker.noliteral.qual.MaybeConstant;
+import org.checkerframework.checker.noliteral.qual.NonConstant;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.basetype.BaseTypeVisitor;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.framework.type.AnnotatedTypeParameterBounds;
+import org.checkerframework.framework.type.TypeHierarchy;
+import org.checkerframework.javacutil.ElementUtils;
+
+/**
+ * This visitor has a lot of code in it that permits this type system to operate as expected with
```

Comment:
Should you create a new class `BottomDefaultTypeVisitor`, and have both systems use it?
That would reduce code, avoid cut-and-paste errors, and ensure that fixes propagate to all implementations.

Reply:
The Checker Framework probably should provide such a class. Let's plan on writing one and getting it reviewed after the FSE deadline.